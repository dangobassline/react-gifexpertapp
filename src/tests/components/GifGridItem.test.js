import React from 'react';
//import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import GifGridItem from '../../components/GifGridItem';


describe('Test of <GifGridItem />', () => {

    const title = 'Un Titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={url}/> );


    test('should mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should de tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(title);
    });

    test('should de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('should de tener el animated__fadeIn', () => {
        const div = wrapper.find('div');
        expect(div.hasClass('animate__fadeIn')).toBeTruthy();
    })


});