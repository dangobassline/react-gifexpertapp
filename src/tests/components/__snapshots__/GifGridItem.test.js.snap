// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <GifGridItem /> should mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un Titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un Titulo
  </p>
</div>
`;
